app/controllers/simplified-account/settings/card-types/card-types.controller.test.js

```diff
@@ -0,0 +1,131 @@
+const sinon = require('sinon')
+const { expect } = require('chai')
+const User = require('@models/User.class')
+const userFixtures = require('@test/fixtures/user.fixtures')
+const proxyquire = require('proxyquire')
+
+const ACCOUNT_TYPE = 'live'
+const SERVICE_ID = 'service-id-123abc'
+
+const adminUser = new User(userFixtures.validUserResponse({
+  external_id: 'user-id-for-admin-user',
+  service_roles: {
+    service: {
+      service: { external_id: SERVICE_ID },
+      role: { name: 'admin' }
+    }
+  }
+}))
+const viewOnlyUser = new User(userFixtures.validUserResponse(
+  {
+    external_id: 'user-id-for-view-only-user',
+    service_roles: {
+      service:
+        {
+          service: { external_id: SERVICE_ID },
+          role: { name: 'view-only' }
+        }
+    }
+  }))
+
+const allCardTypes = [{
+  id: 'id-001',
+  brand: 'visa',
+  label: 'Visa',
+  type: 'DEBIT',
+  requires3ds: false
+},
+{
+  id: 'id-002',
+  brand: 'visa',
+  label: 'Visa',
+  type: 'CREDIT',
+  requires3ds: false
+}]
+
+let req, res, responseStub, getAllCardTypesStub, getAcceptedCardTypesForServiceAndAccountTypeStub, cardTypesController
+
+const getController = (stubs = {}) => {
+  return proxyquire('./card-types.controller', {
+    '@utils/response': { response: stubs.response },
+    '@services/card-types.service': {
+      getAllCardTypes: stubs.getAllCardTypes,
+      getAcceptedCardTypesForServiceAndAccountType: stubs.getAcceptedCardTypesForServiceAndAccountType
+    }
+  })
+}
+
+const setupTest = (method, user, additionalReqProps = {}, additionalStubs = {}) => {
```

Comment:
can you refactor this test to use the new ControllerTestBuilder?

Reply:
yes!